operator/src/main/java/com/instaclustr/k8s/watch/WatchService.java

```diff
@@ -0,0 +1,210 @@
+        this.resourceCache = resourceCache;
+        this.keyFunction = keyFunction;
+        this.resourceMetadataFunction = resourceMetadataFunction;
+        this.resourceListMetadataFunction = resourceListMetadataFunction;
+        this.resourceListItemsFunction = resourceListItemsFunction;
+
+        this.gson = apiClient.getJSON().getGson();
+
+
+        apiClient.getHttpClient().setReadTimeout(1, TimeUnit.MINUTES);
+    }
+
+    @Override
+    protected String serviceName() {
+        return WatchService.class.getSimpleName() + "(" + resourceType.getRawType().getSimpleName() + ")";
+    }
+
+    @Override
+    protected void run() throws Exception {
+        while (isRunning()) {
+            final String listResourceVersion = syncResourceCache();
+
+            watchResourceList(listResourceVersion);
+        }
+    }
+
+    private void watchResourceList(final String listResourceVersion) throws ApiException {
+        logger.debug("Watching resource list.");
+
+        try {
+            currentCall = listCallProvider.get(listResourceVersion, true);
+
+            final Watch<JsonObject> watch = Watch.createWatch(apiClient, currentCall, new TypeToken<Watch.Response<JsonObject>>() {}.getType());
+
+            for (final Watch.Response<JsonObject> objectResponse : watch) {
+                final ResponseType responseType = ResponseType.valueOf(objectResponse.type);
+
+                if (responseType == ResponseType.ERROR) {
+                    final V1Status status = gson.fromJson(objectResponse.object, new TypeToken<Watch.Response<V1Status>>() {}.getType());
+
+                    logger.error("Resource list watch failed with {}.", status);
+
+                    return; // force re-sync
+                }
+
+                final ResourceT resource = gson.fromJson(objectResponse.object, resourceType.getType());
+                final ResourceKeyT key = keyFunction.apply(resource);
+
+                switch (responseType) {
+                    case ADDED:
+                        addCachedResource(key, resource);
+                        break;
+
+                    case MODIFIED:
+                        updateCachedResource(key, resource);
+                        break;
+
+                    case DELETED:
+                        removeCachedResource(key);
+                        break;
+                }
+            }
+
+        } catch (final RuntimeException e) {
+            final Throwable cause = e.getCause();
+
+            if (cause instanceof java.net.SocketTimeoutException) {
+                return; // restart the watch by forcing a re-sync
+            }
+
+            throw e;
+        }
+    }
+
+    private String syncResourceCache() throws ApiException {
+        logger.debug("Synchronising local cache.");
+
+        final ResourceListT apiResponseData;
+        {
+            final ApiResponse<ResourceListT> apiResponse = apiClient.execute(listCallProvider.get(null, false), resourceListType.getType());
```

Comment:
io.kubernetes.client.ApiException: Not Found (404)
Request{method=GET, url=https://192.168.64.14:8443/apis/stable.instaclustr.com/v1/namespaces/default/cassandra-clusters?watch=false, tag=null}

Response:
404 page not found
{"apiVersion":"stable.instaclustr.com/v1","items":[],"kind":"","metadata":{"continue":"","resourceVersion":"47967","selfLink":"/apis/stable.instaclustr.com/v1/namespaces/default/cassandra-clusters"}}

Reply:
v1.9.4 (minikube 0.25.2)

Reply:
Tried on a fresh cluster as well, no luck